### ISSUE TITLE
Storage the Token types of search parameters.

### ISSUE THREAD

**Issue by whyfate** [FirelyTeam/spark#328]
When i read the `Spark`'s source code,I found that the `code` property was added when saved the `searchindex` of the `boolean` type,like this
```json
{
   "active":{
        "code": true 
     }
}
```
https://github.com/FirelyTeam/spark/blob/stu3/master/src/Spark.Engine/Search/ElementIndexer.cs#L354
and I read the fhir's doc
http://build.fhir.org/search.html#token
Token search parameters are used for the following data types:
`boolean`、`code`、`uri`、`Coding` and soon.

but the  `code`、`uri` not add `code` property like `boolean`?
https://github.com/FirelyTeam/spark/blob/stu3/master/src/Spark.Engine/Search/ElementIndexer.cs#L190
https://github.com/FirelyTeam/spark/blob/stu3/master/src/Spark.Engine/Search/ElementIndexer.cs#L447
As a result, when querying Token type, a `plainStringQueries` is required.
https://github.com/FirelyTeam/spark/blob/stu3/master/src/Spark.Mongo/Search/Searcher/CriteriaMongoExtensions.cs#L360

It's a question for me.
Could give me an answer,thinks.

**Comment by kennethmyhra** [comment]
Good catch, @whyfate, seems like the indexing for token search parameters should be consolidated so that we handle all token search parameters similar independent of their underlying data type.

**Comment by kennethmyhra** [comment]
What's your thoughts on this @whyfate, is this something that is time critical to you at the moment? I'm guessing not based on the date for this issue.

If not I will put it into the 2.0 release and then we'll reconsider it then

**Comment by whyfate** [comment]
It's not important at the moment.
